data/state/accountsDB.go

```diff
@@ -600,6 +600,24 @@ func (adb *AccountsDB) IsPruningEnabled() bool {
 	return adb.mainTrie.IsPruningEnabled()
 }
 
+// GetAllLeaves returns all the leaves from a given rootHash
+func (adb *AccountsDB) GetAllLeaves(rootHash []byte) (map[string][]byte, error) {
```

Comment:
I think you should do a merge from economics to rating first. Some of these changes having been already reviewed.

Reply:
fixed